src/encoding.md

```diff
@@ -344,6 +345,56 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-invalid-csv]: #ex-invalid-csv
+<a name="ex-invalid-csv"></a>
+## Handle invalid CSV data with Serde
+
+[![csv-badge]][csv] [![cat-encoding-badge]][cat-encoding]
+
+We get deserialization errors if invalid CSV data like empty field
+is encountered. Serde converts any such errors to a None value, when applied
+to this Option field.
+
+
+```rust
+extern crate csv;
+extern crate serde;
+#[macro_use]
+extern crate serde_derive;
+
+use std::error::Error;
+use std::process;
+
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "PascalCase")]
+struct Record {
+    Name: String,
+    #[serde(deserialize_with = "csv::invalid_option")]
+    Place: Option<String>,
+    id: u64,
+}
+
+fn run() -> Result<(), Box<Error>> {
+    let data = "\
+        Name,Place,id
+        Mark,,46
```

Comment:
this field is actually valid csv the place will be an empty string
please see the https://docs.rs/csv/1.0.0-beta.3/csv/fn.invalid_option.html for suggestion on invalid data from serde point of view.